dspace-api/src/main/java/org/dspace/xmlworkflow/cristin/LinkCollectionToWorkflow.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * The contents of this file are subject to the license and copyright detailed
+ * in the LICENSE and NOTICE files at the root of the source tree and available
+ * online at
+ * <p>
+ * http://www.dspace.org/license/
+ */
+package org.dspace.xmlworkflow.cristin;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.log4j.Logger;
+import org.dspace.content.Collection;
+import org.dspace.content.Community;
+import org.dspace.content.factory.ContentServiceFactory;
+import org.dspace.core.Context;
+import org.dspace.harvest.HarvestedCollection;
+import org.dspace.harvest.factory.HarvestServiceFactory;
+import org.dspace.harvest.service.HarvestedCollectionService;
+import org.dspace.services.ConfigurationService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ *
+ * @author ra
+ *
+ * prerequisite: Only the existing CRIStin and Inspera harvesting collections are mapped.
+ *
+ * This class reads the CRIStin and the Inspera harvesting collections from the database and
+ * mapps the collections handle to the xmlworkflow named 'cristin' resp. 'inspera' The mapping
+ * is stored in the config-file config/workflow.xml
+ *
+ *
+ */
+public class LinkCollectionToWorkflow {
+
+    private static final Logger log = Logger.getLogger(LinkCollectionToWorkflow.class);
+    private static HarvestedCollectionService harvestedCollectionService;
+    private static ConfigurationService configurationService;
+    private static Context context;
+
+    public LinkCollectionToWorkflow() {
+        harvestedCollectionService = HarvestServiceFactory.getInstance().getHarvestedCollectionService();
+        configurationService = DSpaceServicesFactory.getInstance().getConfigurationService();
+        context = new Context();
+    }
+
+    public static void main(String[] args) {
+        configurationService = DSpaceServicesFactory.getInstance().getConfigurationService();
+        harvestedCollectionService = HarvestServiceFactory.getInstance().getHarvestedCollectionService();
+        context = new Context();
+        final String workflowFileName = configurationService.getProperty("dspace.dir") + File.separator +
+                "config" + File.separator + "workflow.xml";
+        StringBuilder stringBuilder = new StringBuilder();
+        try {
+            generateMapping(stringBuilder);
+            FileInputStream inputStream = new FileInputStream(workflowFileName);
```

Comment:
**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *inputStream*. The resource is referenced in statements at the following line: 66. The resource closure statement is at line: 67. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by IOUtils.toString. Either a) close *inputStream* in a try-finally block or b) close the resource by declaring *inputStream* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).